evm/rlp/headers.py

```diff
@@ -112,7 +112,10 @@ def from_parent(cls,
                     timestamp,
                     coinbase=ZERO_ADDRESS,
                     nonce=None,
-                    extra_data=None):
+                    extra_data=None,
+                    transaction_root=None,
+                    receipt_root=None,
+                    state_root=None):
```

Comment:
I'm curious about these additions.  They don't  appear to be used anywhere in this PR.

Reply:
@pipermerriam Since the default empty root is `BLANK_ROOT_HASH`, these additions will be used in `ShardingVM.create_header_from_parent` to use `EMPTY_SHA3` as the empty root:

```python
def create_sharding_header_from_parent(parent_header, **header_params):
    if 'transaction_root' not in header_params:
        header_params['transaction_root'] = EMPTY_SHA3
    if 'receipt_root' not in header_params:
        header_params['receipt_root'] = EMPTY_SHA3
    return create_byzantium_header_from_parent(parent_header, **header_params)
```

And after you pointed out, I found that I shouldn't add `state_root=None`, it shouldn't be overwritten here. Will remove `state_root` parameter, thank you!